commit eb5c02df634b34ca45ecbf1eaf4440ccf806845f
Author: Brad Hubbard <bhubbard@redhat.com>
Date:   Wed May 17 13:28:26 2017 +1000

    mgr: Mark session connections down on shutdown
    
    Also assert in OSD::ms_get_authorizer since it is dangerous to proceed
    if we are shutting down.
    
    Fixes: http://tracker.ceph.com/issues/19900
    
    Signed-off-by: Brad Hubbard <bhubbard@redhat.com>

diff --git a/src/mgr/MgrClient.cc b/src/mgr/MgrClient.cc
index 2f558c0614a..45d2707af25 100644
--- a/src/mgr/MgrClient.cc
+++ b/src/mgr/MgrClient.cc
@@ -59,7 +59,10 @@ void MgrClient::shutdown()
   command_table.clear();
 
   timer.shutdown();
-  session.reset();
+  if (session) {
+    session->con->mark_down();
+    session.reset();
+  }
 }
 
 bool MgrClient::ms_dispatch(Message *m)
